### DISCUSSION TITLE
Is it possible to disbale a input field

### DISCUSSION THREAD

**Discussion by tjoskar** [react-hook-form/react-hook-form#4295] [Q&A] [answered]
Hi 👋 

Let's say we have a 2-step form, where I want to show the second input after the first is submitted. Something like this:
 ```tsx
function App() {
  const { register, handleSubmit, errors } = useForm();
  const [showExtraInput, setShowExtraInput] = React.useState(false);
  const onSubmit = data => {
    setShowExtraInput(true);
    console.log(data);
  };

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <label>Example1</label>
      <input name="example1" ref={register} disabled={showExtraInput} />
      {showExtraInput && (
        <>
          <label>Example2</label>
          <input name="example2" ref={register} />
        </>
      )}
      
      <input type="submit" />
    </form>
  );
}
```

Now, let's say the user enters "123" in the first input field and hit the submit button. `{ example1: "123" }` will be printed in the console (as expected) and the second input is shown. But if the user now enter `456` in the second input field and clicks on submit `{example1: undefined, example2: "456"}` is printed. I expected `{example1: "123", example2: "456"}`. Is this a bug or expected behavior?

I get the expected output (`{example1: "123", example2: "456"}`) if I remove `disabled={showExtraInput}`.

CodeSandbox: https://codesandbox.io/s/react-hook-form-hide-and-disable-input-68cvd?file=/src/index.js

Thanks,

**Comment by bluebill1049** [comment, marked as answer]
you want `readOnly`, instead of `disabled`.

**Reply by tjoskar** [reply]
🤦
– Thanks!
